src/Ardalis.Specification/SpecificationEvaluator.cs

```diff
@@ -47,7 +46,7 @@ public static async Task<IQueryable<T>> GetQuery(IQueryable<T> inputQuery, ISpec
                 query = query.Skip(specification.Skip)
                              .Take(specification.Take);
             }
-            return query;
+            return Task.FromResult(query);
```

Comment:
Did you modify this due to a compiler warning, or ?

Reply:
@ardalis Yes, that's correct. Since the method does not do any actual async operation, call and be still synchronous operation.

Reply:
True. In your opinion is there value in returning a task at all, instead of just returning `IQueryable<T>` ?

Reply:
Well, bases on the scenario below, yes its recommended. 

In REST API call, has some async calls to get a network or database request and this case all other method making to async / await is beneficial since it won't make the incoming thread to wait to process a synchronous call.  And making the whole request pipeline is async/await is recommended.  (https://github.com/davidfowl/AspNetCoreDiagnosticScenarios/blob/master/AsyncGuidance.md)

If we don't require any asyc/await call for the method we can simply return IQueryable<T>. Since this library using some places and I don't recommend to make a breaking change to async to sync.

Reply:
This method originally was synchronous. The method should do only in-memory operations, it just evaluates different expressions and translates into iqueryable. Any async calls to external libraries within the evaluator might affect overall performance (specs should be created on the fly). Externall calls should be responsibility of the repository.

Reply:
That's correct, adding a "async" will generate extra code on the fly. Please check both links with 'async' and with 'no async'

With Async = https://sharplab.io/#v2:D4AQTAjAsAUCAMACEECsBuWsQGZlkQGFEBvWRC5PEAFkQFkAKASlPMoF92LvEAHAE4BLAG4BDAC4BTZBABsyABzIFACSkAbDQHsA6toEaAJgEEAzgE8AdgGMA+gFEAHhIFiWZGJW+JeFEACcKgB0AGIC2gC2AEpSZgCuGhKMAETqWtqIAO4GxinM6JS8XDAcQA==

With out Async = https://sharplab.io/#v2:D4AQTAjAsAUCAMACEECsBuW55bAZkQG9ZFTkwwiSyaUBOACgCJAeDcBB9pgSkxhoF9qpQYgAOAJwCWANwCGAFwCmyCADZkagBIKANtoD2AdT1jtAEwCCAZwCeAOwDGDTsV40yw0iADs6gHQAxMT0AWwAlBUsAV205Zi1dPUQAd2MzLnR3V1IBGD4gA==

My proposal was to get rid of async usage, so we don't want to create any unnecessary allocations..  And I really don't see much values to make this method as async.. since its consuming other projects this would be a breaking change as I mentioned above.

Reply:
Right I'm thinking it should just be synchronous and return IQueryable which eliminates all of the async/Task crud from it (and as noted it doesn't actually do any async work).

Reply:
Awesome... If you want I can take care of it as a new PR once this gets approved or merged!

Reply:
Changing Async to Sync will eventually affect following files.. 

![image](https://user-images.githubusercontent.com/554039/84165551-764da000-aa91-11ea-8d24-5eca8fa2cf49.png)

Reply:
If you don't mind, go ahead and make the change to sync. We don't need any of that to be async at this point - it was changed while trying to get GroupBy working with EF Core 3.x.

Reply:
Sure no problem... I will commit the changes now

Reply:
@ardalis, I just committed my changes. Please review https://github.com/ardalis/Specification/pull/21/commits/9bdca6be7a61da1df540cad521a5cf281dfa78db